src/panel/context/Explorer/index.tsx

```diff
@@ -30,17 +30,23 @@ export const ExplorerProvider: FC = ({ children }) => {
   >(undefined);
 
   useEffect(() => {
-    return addMessageHandler((o: DevtoolsExchangeOutgoingMessage) => {
-      if (o.type === "disconnect") {
+    return addMessageHandler((e: DevtoolsExchangeOutgoingMessage) => {
+      if (e.type === "disconnect") {
         setOperations({});
         return;
       }
 
-      if (o.type === "response" && o.data.data) {
+      if (e.type !== "debug") {
```

Comment:
If we add this, we should be able to completely remove the second conditional on line 45

```suggestion
      if (e.type !== "debug" || e.data.type !== "update") {
```

I can't see a reason why `debugEvent.data` would be undefined so I think we're safe to omit that.